### DISCUSSION TITLE
packaging GDL 1.0.0

### DISCUSSION THREAD

**Discussion by slayoo** [gnudatalanguage/gdl#1070] [General]
Dear @gnudatalanguage/gdlpackagers,
CC: @gnudatalanguage/gdlteam

GDL 1.0.0 is out!

The release notes (https://github.com/gnudatalanguage/gdl/releases/tag/v1.0.0) include a brief summary of what have been added to GDL in over 700 commits since the previous release, as well as several notes to packagers.

In particular, we would like to point out that since 1.0.0, GDL uses a different plplot driver for plotting. With a great majority of GDL users potentially interested in plotting features, it would be really great to ensure (through relevant dependency metadata for the packages), that newly installed GDL packages provide users with a working environment in which plplot is equipped with the `wxwidgets` driver.

Please provide here any general feedback regarding packaging of GDL 1.0.0.
Please report any bugs or feature requests as new issues (or PRs!).

Thank you,
GDL Team

**Comment by dirteat** [comment]
Yep, since 1.0.0-rc* the build is broken on arm 32 bits :-/

https://github.com/gnudatalanguage/gdl/issues/677

If this is no longer supported maybe we should mention it in the release notes!
thanks.

**Reply by alaingdl** [reply]
Hi. Let's be clear : if you express a clear wish, we may try to fix/improve.

Comments : 
1/ maybe you can try by cutting most of the options at the beginning. Just to check if it is related
to core GDL code, or due to dependencies ?
2/ Around I have only x86_64 and it is quite difficult to have VM on some configuration.)
3/ I am still fighting to compile GDL on some OSX where I cannot make updates (xcode side) (because the end-user of the machine is not OK for that) and I do consider it is quite important to clarify that.

**Reply by olebole** [reply]
The main point for me (Debian) is also the broken compilation on 32-bit ARM (#677). Since this seems a bug, it would be rather unfortunate to exclude this architecture as "unsupported"; I however also cannot contribute to fix it. I have a Raspberry where I could try to give some (slow) access to if that helps.

**Reply by slayoo** [reply]
Do I understand correctly, from @GillesDuvert's and others' comments in #677 that:
- we use non-standard C++ multiple symbol definitions that most linkers are OK, but with arm32 ABI requirements it fails
- we do it to significantly speed up the builds (4x) and would prefer to keep this speed up available
- the solution is straightforward for arm32 - to include the relevant definitions to one long-compiling file through some preprocessor switch
- a tricky bit is to detect in CMake if a compiler chokes on it or not.

Any hints if someone has already done an analogous solution in another project?

**Reply by olebole** [reply]
There is a lengthy discussion in https://github.com/mamedev/mame/issues/3639 which had the same problem, followed by the PR https://github.com/mamedev/mame/pull/3715 with the solution. Not sure whether this helps here.

**Reply by pjb7687** [reply]
> 3/ I am still fighting to compile GDL on some OSX where I cannot make updates (xcode side) (because the end-user of the machine is not OK for that) and I do consider it is quite important to clarify that.

Regarding OSX packaging, I was actually considering a no-compile installation via `dmg`: https://github.com/gnudatalanguage/gdl/actions/runs/1096161342

Would it be possible for you to give it a try? You can download the artifact, open the dmg file, drag-and-drop the `GNU Data Language.app` to `Applications`, and it should work.

**Reply by alaingdl** [reply]
OK, but not sure to be able to try before tomorrow evening or Tuesday

and quite busy now by my issues on "old" debian/ubuntu no-go pbs :(

**Reply by slayoo** [reply]
I've just tried it on my brand new UIUC laptop and here's the first problem:

```
sarabas@atmos-sarabas ~ % /Applications/GNU\ Data\ Language.app/Contents/Resources/bin/gdl
dyld: Library not loaded: /usr/local/opt/libpng/lib/libpng16.16.dylib
  Referenced from: /Applications/GNU Data Language.app/Contents/Frameworks/libwx_osx_cocoau_core-3.1.dylib
  Reason: image not found
zsh: abort      /Applications/GNU\ Data\ Language.app/Contents/Resources/bin/gdl
```

**Reply by pjb7687** [reply]
Regarding the macOS dmg release, here is another try: https://github.com/gnudatalanguage/gdl/actions/runs/1156851470.
Also started a PR: https://github.com/gnudatalanguage/gdl/pull/1074

**Reply by slayoo** [reply]
thanks @pjb7687! Now, a similar problem occurs wrt MPI:
```
sarabas@atmos-sarabas ~ % /Applications/GNU\ Data\ Language.app/Contents/Resources/bin/gdl
  GDL - GNU Data Language, Version 1.0.0
- For basic information type HELP,/INFO
- Default library routine search path used (GDL_PATH/IDL_PATH env. var. not set): /Users/runner/work/gdl/install/share/gnudatalanguage/lib
- Using WxWidgets as graphics library (windows and widgets).
- No startup file read (GDL_STARTUP/IDL_STARTUP env. var. not set). 
- Please report bugs, feature or help requests and patches at: https://github.com/gnudatalanguage/gdl

--------------------------------------------------------------------------
Sorry!  You were supposed to get help about:
    opal_init:startup:internal-failure
But I couldn't open the help file:
    /usr/local/Cellar/open-mpi/4.1.1_2/share/openmpi/help-opal-runtime.txt: No such file or directory.  Sorry!
--------------------------------------------------------------------------
--------------------------------------------------------------------------
Sorry!  You were supposed to get help about:
    orte_init:startup:internal-failure
But I couldn't open the help file:
    /usr/local/Cellar/open-mpi/4.1.1_2/share/openmpi/help-orte-runtime: No such file or directory.  Sorry!
--------------------------------------------------------------------------
--------------------------------------------------------------------------
Sorry!  You were supposed to get help about:
    mpi_init:startup:internal-failure
But I couldn't open the help file:
    /usr/local/Cellar/open-mpi/4.1.1_2/share/openmpi/help-mpi-runtime.txt: No such file or directory.  Sorry!
--------------------------------------------------------------------------
*** An error occurred in MPI_Init
*** on a NULL communicator
*** MPI_ERRORS_ARE_FATAL (processes in this communicator will now abort,
***    and potentially your MPI job)
[atmos-sarabas:34349] Local abort before MPI_INIT completed completed successfully, but am not able to aggregate error messages, and not able to guarantee that all other processes were killed!
```

**Comment by jaymurthy** [comment]
I tried it. 

1. Double-clicking on the App didn't seem to do anything.2. cd into the App Directory (cd /Applications/GNUDataLanguage.app/Contents/MacOS/) and then typing gdl works.3. starting from outside there gives problems, first with the spaces. I removed the spaces and got this error:dyld: Library not loaded: /usr/local/opt/libpng/lib/libpng16.16.dylib
 Referenced from: /Applications/GNUDataLanguage.app/Contents/Frameworks/libwx_osx_cocoau_core-3.1.dylib
   
 Hope this helps.Jayant
On Monday, August 23, 2021, 02:38:43 AM GMT+5:30, Jeongbin Park ***@***.***> wrote:  
 
 



3/ I am still fighting to compile GDL on some OSX where I cannot make updates (xcode side) (because the end-user of the machine is not OK for that) and I do consider it is quite important to clarify that.


Regarding OSX packaging, I was actually considering a no-compile installation via dmg: https://github.com/gnudatalanguage/gdl/actions/runs/1096161342

Would it be possible for you to give it a try? You can download the artifact, open the dmg file, drag-and-drop the GNU Data Language.app to Applications, and it should work.


—
You are receiving this because you are on a team that was mentioned.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Reply by GillesDuvert** [reply]
Putting the 'gdl app' in front when it is started from 'terminal' (or equivalent) is not possible [unless the gdl executable is an 'app bundle' itself](https://www.wxwidgets.org/docs/faq/osx/#nofocus), something not entirely automated by our building script right now. (perso I've just created and empty bundle directory chain `gdl.app/Contents/MacOS/gdl` where `gdl` is a link to 'where/gdl/is:installed' and it's OK). With this the windows (wigets and plot) can get focus.
For a linux/mageia/kde user, the Apple desktop management is, eh, kind of frustrating...
Second, the widgets will not work well until[ this patch](https://github.com/gnudatalanguage/gdl/pull/1054) is merged, as the default (linux) 'style' of the widget was not accepted by Apple.
I hope we'll provide DMGs soon folowing your advice. 

For the record, with the current version of GDL using wxWidgets (wxWidgets on mac is just a layer atop Cocoa), and our graphic event loop (plots and widgets) is the Apple Desktop eventloop itself (using wxYield()). No specific thread for gdl's graphic events (until somebody with wxWidgets thread cleverness revises our code, let's pray), and this may explain some peculiarities seen.
